docs/readmes/README.best-practices.md

```diff
@@ -0,0 +1,206 @@
+
+Note that there is a brief window between when your application started running and the identity gets assigned to the underlying node. If your application tried to retrieve a token during that period, it will fail. The following table describes the duration that you could expect based on your cluster's node type:
+
+| Node Type                 | Identity Assignment Duration |
+|---------------------------|------------------------------|
+| Virtual Machine           | 10 - 20 seconds              |
+| Virtual Machine Scale Set | 40 - 60 seconds              |
+
+Without retry, your application will not able to retrieve AAD tokens because the identity assignment operation is still in progress. Usually, retry logic is already implemented internally in most of Azure's SDKs to prevent your application from erroring out. For example, [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go) by default performs five retries with exponential backoff when fetching AAD token from IMDS.
+
+Alternatively, you could set up a simple init container to probe for successful identity assignment by MIC and prevent applications from starting immaturely. Here is an example init-container with Azure CLI.
+
+```yaml
+...
+initContainers:
+- name: init
+  image: mcr.microsoft.com/azure-cli
+  command:
+    - sh
+    - -c
+    - az login --identity --debug
+...
+```
+
+## A pod using multiple AzureIdentities
+
+In some scenarios, you might want to assign multiple `AzureIdentities` to a workload. Here is an example of how to achieve that:
+
+### AzureIdentities
+
+```yaml
+apiVersion: aadpodidentity.k8s.io/v1
+kind: AzureIdentity
+metadata:
+  name: az-id-1
+spec:
+  type: 0
+  resourceID: <ResourceID of az-id-1>
+  clientID: <ClientID of az-id-1>
+```
+
+```yaml
+apiVersion: aadpodidentity.k8s.io/v1
+kind: AzureIdentity
+metadata:
+  name: az-id-2
+spec:
+  type: 0
+  resourceID: <ResourceID of az-id-2>
+  clientID: <ClientID of az-id-2>
+```
+
+### AzureIdentityBinding
+
+```yaml
+apiVersion: aadpodidentity.k8s.io/v1
+kind: AzureIdentityBinding
+metadata:
+  name: az-id-1-binding
+spec:
+  azureIdentity: az-id-1-binding
+  selector: az-id-combined
+```
+
+```yaml
+apiVersion: aadpodidentity.k8s.io/v1
+kind: AzureIdentityBinding
+metadata:
+  name: az-id-2-binding
+spec:
+  azureIdentity: az-id-2-binding
+  selector: az-id-combined
+```
+
+### Pod
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: demo
+  labels:
+    aadpodidbinding: az-id-combined
+...
+```
+
+## Pods using unauthorized AzureIdentities
+
+By default, aad-pod-identity matches pods with `AzureIdentities` across all namespaces. That means that a malicious pod could assign itself an unauthorized pod identity label and acquire a token with that particular `AzureIdentity`. This scenario can be mitigated by performing the following:
+
+- Deploy aad-pod-identity with [namespaced mode](README.namespaced.md). This will restrict pods from binding with `AzureIdentities` across different namespaces.
```

Comment:
Lets add a note that in namespaced mode, the pod, AzureIdentity and AzureIdentityBinding all need to be in the same namespace